src/main/java/Course.java

```diff
@@ -1,47 +1,145 @@
+import java.lang.reflect.Array;
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
 
 /** This class represents a course, which consists of lecture and tutorial sessions */
 public class Course {
+
     private final String courseId;
-    private final ArrayList<Session> lectures;
-    private final ArrayList<Session> tutorials;
+    private final Map<String, ArrayList<Session>> lectures;
```

Comment:
Imo should leave this as type `List` rather than `ArrayList`. Knowing it stores `ArrayList` isn't necessary for how it's used in this class as far as I can tell, so may as well stay on higher level of abstraction. Same applies for everywhere ArrayList is explicitly stated as return type or argument type

Reply:
Also what strings are being mapped to Sessions? If it makes sense to do so, it would be nice if that's reflected in the variable name. 

So if it was mapping Course IDs to sessions for example, something like`lecturesByCourseIDs`, just cause it's confusing when reading code later

Reply:
The strings being mapped to sessions are the section ids. As an example,

"LEC0101" -> 'Session for 9am Monday', 'Session for 9am Wednesday', 'Session for 9am Friday'

The advantage to this is there is no looping required when we want to collect all sessions corresponding to one section. (There should probably have been a getter method for this my bad).  In scheduler we would be able to loop through all sections and add all sessions from that section using this map.

Good point about the List vs. Arraylist, I will go back and fix that.